pulsar4s-core/src/test/scala/com/sksamuel/pulsar4s/FutureAsyncHandlerTest.scala

```diff
@@ -49,4 +54,38 @@ class FutureAsyncHandlerTest extends AnyFunSuite with Matchers with BeforeAndAft
     zipped.foreach(t => t._1 shouldBe t._2)
     consumer.close()
   }
+
+  test("async producer and consumer can participate in transaction") {
+    val producer = client.producer(ProducerConfig(topic, sendTimeout = Some(Duration.Zero)))
+    val consumer = client.consumer(ConsumerConfig(topics = Seq(topic), subscriptionName = Subscription("mysub_" + UUID.randomUUID)))
+    consumer.seekEarliest()
+    val msgIdFt = client.transaction { txn =>
+      for {
+        msg <- consumer.receiveAsync
+        msgId <- txn(producer).sendAsync(msg.value + "_test")
+        _ <- txn(consumer).acknowledgeAsync(msg.messageId)
```

Comment:
I like this syntax but this instantiates the wrapper every single call. performance-wise it's negligible, but it seems slightly weird to re-instantiate the wrapper every single time

Reply:
or maybe we're meant to do `txnProducer = txn(producer)` for multiple calls?

Reply:
Yeah, I think whether you use an intermediate variable or not is down to personal preference. I was leaning toward just using the extra parameter like `producer.sendAsync(value, txn)`, which has the advantage of being simpler to implement, but I sort of like the `txn` coming first to make it clear the work is done in a transaction.